packages/plugin-table/src/TableCellNode.ts

```diff
@@ -0,0 +1,170 @@
+    attributes: TableCellAttributes;
+    header: boolean;
+    // Only the `managerCell` setter should modify the following private keys.
+    _managerCell: TableCellNode;
+    _managedCells = new Set<TableCellNode>();
+
+    constructor(header = false) {
+        super();
+        this.header = header;
+    }
+
+    //--------------------------------------------------------------------------
+    // Lifecycle
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return a new VNode with the same type and attributes as this VNode.
+     *
+     *  @override
+     */
+    clone(): this {
+        const clone = new this.constructor<typeof TableCellNode>(this.header);
+        clone.attributes = { ...this.attributes };
+        return clone;
+    }
+
+    //--------------------------------------------------------------------------
+    // Getters
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    get name(): string {
+        let coordinatesRepr = ' <(' + this.rowIndex + ', ' + this.columnIndex + ')';
+        if (this.colspan > 1 || this.rowspan > 1) {
+            const endRow = this.rowIndex + this.rowspan - 1;
+            const endColumn = this.columnIndex + this.colspan - 1;
+            coordinatesRepr += ':(' + endRow + ', ' + endColumn + ')';
+        }
+        coordinatesRepr += '>';
+        return (
+            super.name +
+            coordinatesRepr +
+            (this.header ? ': header' : '') +
+            (this.isActive ? '' : ' (inactive)')
+        );
+    }
+    /**
+     * Return the cell that manages this cell, if any.
+     */
+    get managerCell(): TableCellNode {
+        return this._managerCell;
+    }
+    /**
+     * Set the given cell as manager of this cell. If null is passed, this cell
+     * will have no manager.
+     */
+    set managerCell(newManager: TableCellNode | null) {
+        const oldManager = this._managerCell;
+        this._managerCell = newManager;
+
+        if (newManager) {
+            // If we just set a new manager for this cell, also add this cell to
+            // the new manager's managed cells.
+            newManager._managedCells.add(this);
+
+            // Copy the manager.
+            this.attributes = { ...newManager.attributes };
+            this.header = newManager.header;
+
+            // Move the children to the manager.
+            newManager.append(...this.children);
+
+            // Hand the managed cells over to the manager.
+            for (const managedCell of this.managedCells) {
+                managedCell.managerCell = newManager;
+                this._managedCells.delete(managedCell);
+            }
+        } else if (oldManager) {
+            // If we just removed this cell's manager, also remove this cell
+            // from the old manager's managed cells.
+            oldManager._managedCells.delete(this);
```

Comment:
Now that I think about it, how would the system react if I set a manager on non-contiguous cells ?